### ISSUE TITLE
Consider .hpp and .hxx files to be header files too (C++)

### ISSUE THREAD

**Issue by BrainStone** [gradle/gradle-native#1079]
### Expected Behavior
Consider all 3 common C++ header endings as valid header file endings: `.h`, ``.hpp`` and ``.hxx``

### Current Behavior
Currently the plugin is hardcoded to only use ``.h`` files:
https://github.com/gradle/gradle/blob/ea650c049886424271b3583cc57ba260458f68c0/subprojects/language-native/src/main/java/org/gradle/language/cpp/internal/DefaultCppComponent.java#L112

### Context
Other plugins trying to find the header files (like the VisualStudio plugin) will only get a list of files with the ``.h`` ending when they try to get a list of all header files.

**Comment by lacasseio** [comment]
Thanks for opening this issue. [Nokee](https://nokee.dev) also consider `h++` extension as being C++ headers. However, we also allow this default value to be configurable.

**Comment by BrainStone** [comment]
Yeah you are right. The `.h++` extension should be supported as well.

**Comment by abelavusau** [comment]
Hello,

What's about concidering *.ipp files as well?
